portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,34 +27,44 @@
 public class DataServlet extends HttpServlet {
 
   private List<String> messages;
+  private static final String comment = "comment-input";
+  private static final String name = "name-input";
 
   @Override
   public void init() {
     messages = new ArrayList<>();
-    messages.add("Placeholder message #1.");
-    messages.add("Placeholder message #2.");
-    messages.add("Placeholder message #3.");
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Construct comment from user request.
+    String fullComment = request.getParameter(name) + ": " + request.getParameter(comment);
+
+    response.setContentType("text/html;");
+    response.getWriter().println(fullComment);
+
+    messages.add(fullComment);
```

Comment:
just so all the response code is together, move this to right under the fullComment line.